contract/contracts/AstarBase.sol

```diff
@@ -1,50 +1,88 @@
-    string MSG_PREFIX = "Sign this to register to AstarBase for:";
-    uint256 public unregisterFee = 1 ether;
-    address public beneficiary = 0x91986602d9c0d8A4f5BFB5F39a7Aa2cD73Db73B7; // Faucet on all Astar networks
-
-    constructor() {
+    mapping(address => bytes) public addressMap;
+    mapping(bytes => address) public ss58Map;
+    DappsStaking public DAPPS_STAKING;
+    SR25519 public SR25519Contract;
+
+    // Emitted when the getVersion() is called
+    event ContractVersion(uint256 newValue);
+    event AstarBaseRegistered(address newEntry);
+
+    function initialize() public initializer {
+        __Ownable_init();
+        registeredCnt.reset();
+        version = 0;
+        PREFIX = hex"3c42797465733e";
+        POSTFIX = hex"3c2f42797465733e";
         paused = false;
+        MSG_PREFIX = "Sign this to register to AstarBase for:";
+        unregisterFee = 1 ether;
+        beneficiary = 0x91986602d9c0d8A4f5BFB5F39a7Aa2cD73Db73B7; // Faucet on all Astar networks
+        DAPPS_STAKING = DappsStaking(0x0000000000000000000000000000000000005001);
+        SR25519Contract = SR25519(0x0000000000000000000000000000000000005002);
+    }
+
+    /// @notice Check upgradable contract version.
+    /// @notice Change this version value for each new contract upgrade
+    function getVersion() public {
+
+        emit ContractVersion(1);
     }
 
     /// @notice Register senders' address with corresponding SS58 address and store to mapping
+    /// @notice The accounts authenticity is verified through signature
     /// @param ss58PublicKey, SS58 address used for signing
     /// @param signedMsg, The message that was signed should be constructed as:
     ///                   MSG_PREFIX + ss58PublicKey + msg.sender
-    function register(bytes32 ss58PublicKey, bytes calldata signedMsg) external {
+    function register(bytes calldata ss58PublicKey, bytes calldata signedMsg) external {
         require(!paused, "The contract is paused");
-        require(ss58PublicKey != 0, "Can't register ss58PublicKey with 0");
+        require(ss58PublicKey.length != 0, "Can't register ss58PublicKey with 0");
         require(ss58Map[ss58PublicKey] == address(0), "Already used ss58 Public Key");
-        require(addressMap[msg.sender] == 0, "Already registered evm address");
+        require(addressMap[msg.sender].length == 0, "Already registered evm address");
 
         bytes memory messageBytes = bytes(MSG_PREFIX);
         bytes memory addressInBytes = abi.encodePacked(msg.sender);
-        bytes memory fullMessage = bytes.concat(PREFIX, messageBytes, ss58PublicKey, addressInBytes, POSTFIX);
-        bool address_verified = SR25519Contract.verify(ss58PublicKey, signedMsg, fullMessage);
+        bytes32 pubKey = bytesToBytes32(ss58PublicKey, 0);
+        bytes memory fullMessage = bytes.concat(PREFIX, messageBytes, pubKey, addressInBytes, POSTFIX);
+        bool address_verified = SR25519Contract.verify(pubKey, signedMsg, fullMessage);
```

Comment:
We should also add ECDSA logic. Contract is working fine with correct public key. Only frontend needs to be changed.